daemons/ipa-kdb/ipa_kdb_principals.c

```diff
@@ -505,6 +507,50 @@ static krb5_error_code ipadb_get_ldap_auth_ind(krb5_context kcontext,
     return ret;
 }
 
+static void ipadb_parse_authind_policies(krb5_context kcontext,
+                                         LDAP *lcontext,
+                                         LDAPMessage *lentry,
+                                         krb5_db_entry *entry,
+                                         enum ipadb_user_auth ua)
+{
+    int result;
+    int ret;
+    const struct {
+        char *attribute;
+        enum ipadb_user_auth flag;
+    } life_authind_map[] = {
+        {"krbAuthIndMaxTicketLife;otp", IPADB_USER_AUTH_OTP},
+        {"krbAuthIndMaxTicketLife;radius", IPADB_USER_AUTH_RADIUS},
+        {"krbAuthIndMaxTicketLife;pkinit", IPADB_USER_AUTH_PKINIT},
+        {"krbAuthIndMaxTicketLife;hardened", IPADB_USER_AUTH_HARDENED},
+	{NULL, IPADB_USER_AUTH_NONE},
+    }, age_authind_map[] = {
+        {"krbAuthIndMaxRenewableAge;otp", IPADB_USER_AUTH_OTP},
+        {"krbAuthIndMaxRenewableAge;radius", IPADB_USER_AUTH_RADIUS},
+        {"krbAuthIndMaxRenewableAge;pkinit", IPADB_USER_AUTH_PKINIT},
+        {"krbAuthIndMaxRenewableAge;hardened", IPADB_USER_AUTH_HARDENED},
+        {NULL, IPADB_USER_AUTH_NONE},
+    };
+
+    for(size_t i = 0; life_authind_map[i].attribute != NULL; i++) {
+        if (ua & life_authind_map[i].flag) {
+            ret = ipadb_ldap_attr_to_int(lcontext, lentry,
+                                         life_authind_map[i].attribute,
+                                         &result);
+            if (ret == 0) {
+                entry->max_life = result;
+	    }
+            ret = ipadb_ldap_attr_to_int(lcontext, lentry,
+                                         age_authind_map[i].attribute,
+                                         &result);
+            if (ret == 0) {
+                entry->max_renewable_life = result;
+            }
```

Comment:
Would it make sense to do a break in here since if we have a match of either the life or age map for an indicator then the others would be guaranteed to not match? I guess it's just over 8 items but still.

Reply:
Probably yes, but we have few cases where these are used as flags, so theoretically all of them can be set. It doesn't make sense though...